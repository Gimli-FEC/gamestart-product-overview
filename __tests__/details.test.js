import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';

import Details from '../client/components/details.jsx';

const propsObj = {
  info: {
    id: 1,
    title: 'Practical Wooden Pants',
    publisher: 'Schuppe, Klocko and Kirlin',
    content_rating: 6,
    priceNew: 79.79,
    priceUsed: 39.31,
    currentStockNew: 3,
    currentStockUsed: 2,
  },
  esrb: {
    name: 'adults-only',
    url: 'https://hrr45-fec.s3.us-east-2.amazonaws.com/photos/esrb/6.png',
  },
};

afterEach(cleanup);

test('renders', () => {
  const { asFragment } = render(<Details
    title={propsObj.info.title}
    publisher={propsObj.info.publisher}
    EsrbSrc={propsObj.esrb.url}
    EsrbCat={propsObj.esrb.name}
    priceNew={propsObj.info.priceNew}
    priceUsed={propsObj.info.priceUsed} />);
  expect(asFragment()).toMatchSnapshot();
});
