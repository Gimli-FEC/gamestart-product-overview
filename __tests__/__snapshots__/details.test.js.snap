// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="details__Grid-sc-2uoopj-3 laHZWJ"
  >
    <div
      class="details__Row-sc-2uoopj-4 ecJMdW"
    >
      <div
        class="details__Col-sc-2uoopj-5 hihrbV"
        size="3"
      >
        <div
          class="details__Row-sc-2uoopj-4 ecJMdW"
        >
          <h1
            class="details__Title-sc-2uoopj-0 hLIyKG"
          >
            Practical Wooden Pants
          </h1>
        </div>
        <div
          class="details__Row-sc-2uoopj-4 ecJMdW"
        >
          <span
            class="details__Publisher-sc-2uoopj-1 jXFZJi"
          >
            Schuppe, Klocko and Kirlin
          </span>
        </div>
      </div>
      <div
        class="details__Col-sc-2uoopj-5 cKwRRf"
        size="1"
      >
        <img
          alt="adults-only"
          class="details__EsrbPic-sc-2uoopj-2 hUcYWy"
          src="https://hrr45-fec.s3.us-east-2.amazonaws.com/photos/esrb/6.png"
        />
      </div>
    </div>
    <div
      class="details__Row-sc-2uoopj-4 ecJMdW"
    >
      <div
        class="ratings__Row-sc-8kadw3-3 iEETze"
      >
        <div
          class="ratings__Col-sc-8kadw3-4 cwiHmS"
          size="1"
        >
          <div
            class="ratings__Container-sc-8kadw3-0 cbQoLz"
          >
            <span
              class="ratingsTooltip__Popup-sc-1d6glbo-0 irRflo"
            >
              <span
                class="ratingsTooltip__Trigger-sc-1d6glbo-1 bevQCd"
              >
                <div
                  class="ratings__TopStars-sc-8kadw3-1 dEntyF"
                >
                  ⭑⭑⭑⭑⭑
                </div>
                <div
                  class="ratings__BottomStars-sc-8kadw3-2 iBSdYZ"
                >
                  ⭑⭑⭑⭑⭑
                </div>
              </span>
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="details__Row-sc-2uoopj-4 ecJMdW"
    >
      <div
        class="version__RadioContainer-c6dro4-4 hvzLpD"
      >
        <div
          class="version__LabelContainer-c6dro4-3 bvmExG"
        >
          <div
            class="version__Icon-c6dro4-2 ftBzxF"
          />
          <label
            class="version__Label-c6dro4-0 kKxBiA"
          >
            <strong>
              new
            </strong>
            <br />
            <div
              class="version__ValueDiv-c6dro4-1 kVroOq"
            >
              $79.79
            </div>
            <input
              name="version"
              type="radio"
              value="79.79"
            />
          </label>
        </div>
        <div
          class="version__LabelContainer-c6dro4-3 bvmExG"
        >
          <div
            class="version__Icon-c6dro4-2 ftBzxF"
          />
          <label
            class="version__Label-c6dro4-0 kKxBiA"
          >
            <strong>
              pre-owned
            </strong>
            <br />
            <div
              class="version__ValueDiv-c6dro4-1 kVroOq"
            >
              $39.31
            </div>
            <input
              name="version"
              type="radio"
              value="39.31"
            />
          </label>
        </div>
      </div>
    </div>
    <div
      class="details__Row-sc-2uoopj-4 ecJMdW"
    >
      Available at a store near you!
    </div>
    <div
      class="details__Row-sc-2uoopj-4 ecJMdW"
    >
      <span>
        <strong>
           FREE NO HURRY SHIPPING $35+
        </strong>
      </span>
    </div>
    <div
      class="details__Row-sc-2uoopj-4 ecJMdW"
    >
      <button
        class="button__AddToCart-sc-1vptnfg-0 dVUQAg"
      >
        Add to cart
      </button>
    </div>
    <div
      class="details__Row-sc-2uoopj-4 ecJMdW"
    >
      IN STOCK Component Coming Soon
    </div>
    <div
      class="details__Row-sc-2uoopj-4 ecJMdW"
    >
      <div
        class="details__Col-sc-2uoopj-5 cKwRRf"
        size="1"
      >
        ADD TO WISHLIST
      </div>
      <div
        class="details__Col-sc-2uoopj-5 cKwRRf"
        size="1"
      >
        SEE TRADE VALUE
      </div>
    </div>
  </div>
</DocumentFragment>
`;
